tests/test_validation_mismatch.py

```diff
@@ -0,0 +1,781 @@
+from sphinxval.utils import validation as validate
+from sphinxval.utils import metrics
+from sphinxval.utils import time_profile as profile
+from sphinxval.utils import resume
+from sphinxval.utils import plotting_tools as plt_tools
+from sphinxval.utils import match
+from sphinxval.utils import validation_json_handler as vjson
+from sphinxval.utils import classes as cl
+from sphinxval.utils import object_handler as objh
+
+
+
+
+from astropy import units
+import unittest
+import sys
+import pprint
+import pandas as pd
+import os
+import csv
+import logging
+import logging.config
+import pathlib
+import json
+from datetime import datetime
+
+from unittest.mock import patch
+import shutil # using this to delete the contents of the output folder each run - since the unittest is based on the existence/creation of certain files each loop
+
+
+
+logger = logging.getLogger(__name__)
+
+
+
+"""
+Mismatch unit test requires a slightly different 
+set up from the regular validation unit test
+since the dataframe takes different values
+from the config file (which I mock so that I can
+more easily control the values of). Also since
+this is mismatch the output file names may contain
+_mm at the end 
+"""
+
+
+def utility_setup_logging():
+    # Create the tests/logs/ directory if it does not yet exist
+    if not os.path.exists(config_tests.logpath):
+        os.mkdir(config_tests.logpath)
+
+    config_file = pathlib.Path('tests/log_config_tests.json')
+    with open(config_file) as f_in:
+        config_ = json.load(f_in)
+    logging.config.dictConfig(config_)
+
+
+
+def utility_load_forecast(filename, energy_channel):
+    """
+    Utility function for unit testing. Loads forecast object directly from JSON.
+       Input:
+            :filename: (string) forecast JSON file.
+            :energy_channel: (dict) 
+       Output:
+            :forecast: resultant Forecast() object.
+    """
+    forecast_dict = vjson.read_in_json(filename)
+    forecast, _ = vjson.forecast_object_from_json(forecast_dict, energy_channel)  
+    # obj, _ = vjson.forecast_object_from_json(forecast_dict, config_tests.mm_pred_energy_channel)
+    # forecast.update({config_tests.mm_energy_key: []})
+    # if not pd.isnull(obj.prediction_window_start):
+    #     forecast[config_tests.mm_energy_key].append(obj)
+    #     logger.debug("Adding " + obj.source + " to dictionary under key " + key)  
+    # if config_tests.do_mismatch:
+        
+    #         if channel == config_tests.mm_obs_energy_channel:
+    #             pred_channel = config_tests.mm_pred_energy_channel
+                
+ 
+    #             if not is_good:
+    #                 logger.warning("Note issue with creating FORECAST object from json " + str(obj.source)  + ", mismatch channel" + str(pred_channel))
+
+    #             #skip if energy block wasn't present in json
+                
+    return forecast
+
+def utility_load_observation(filename, energy_channel):
```

Comment:
Duplicated from test_validation.py -- move to common module and import e.g. `tests.utils`